src/main/java/io/cryostat/net/AuthManager.java

```diff
@@ -49,6 +51,10 @@
 
     Future<UserInfo> getUserInfo(Supplier<String> httpHeaderProvider);
 
+    Optional<String> sendLoginRedirectIfRequired(
```

Comment:
I think this name can be simplified a bit. `IfRequired` seems to already be implied by the `Optional` return type. Maybe just `getLoginRedirectUrl`?